streamer/output_stream.py

```diff
@@ -29,7 +32,14 @@ def __init__(self,
                codec: Union[AudioCodec, VideoCodec, None]) -> None:
     self.type: MediaType = type
     # If "pipe" is None, then this will not be transcoded.
-    self.pipe: Optional[str] = pipe
+    self.read_pipe: Optional[str] = pipe
+    self.writ_pipe: Optional[str] = pipe
+    # On posix systems, the read and write pipes will be the same.
+    # But on Windows, we will have different read and write pipe names.
+    if os.name == 'nt' and self.read_pipe and self.writ_pipe:
```

Comment:
If i understood that correctly, you mean to put a pipe before and after each node in the pipe line, like this.
![image](https://user-images.githubusercontent.com/56920956/127730981-dd4e7953-c598-420f-a542-52c940f7ee93.png)

So we would make a new file pipe.py and let all the platform diversion logic be in there, and also the pipe path naming(could be a file).

Reply:
I think we will need only the middle and the rightmost pipe in the blueprint up above, as the `inputs` will represent leftmost pipe.

Reply:
I have invested some time working on removing the 'file' terminology from the `PackagerNode` and replacing it with an output pipe instead, but i am afraid that we might not be able to do that, since the output file is sometimes two and not one, and is a template for the initialization segment and other segments of the media file.
We might though be able to embed these knowledge in the `OutputStream` class instead and let it deal with the pipe creation and file templates too.